File: Python/8py 2/remove_marks2.py
"""
|--------------------------------------------------------------------------
| Exclamation marks series #2: Remove all exclamation marks from the end of sentence
|--------------------------------------------------------------------------
|
"""

def remove(s):
   x, string = 0, list(reversed(s))
   for i in string:
      if i != "!": break
      x += 1
   return s[0:len(s)-x]


def remove(s):
   return s.rstrip("!")


def remove(s):
   while s and s[-1] == "!":
      s = s[:-1]
   return s


from re import sub
def remove(s):
   return sub('[!]+$', '', s)


print(remove("Hi!"), "Hi")
print(remove("Hi!!!"), "Hi")
print(remove("!Hi"), "!Hi")
print(remove("!Hi!"), "!Hi")
print(remove("Hi! Hi!"), "Hi! Hi")
print(remove("Hi"), "Hi")

File: JavaScript/8js 2/remove_marks2.js
/*
|--------------------------------------------------------------------------
| Exclamation marks series #2: Remove all exclamation marks from the end of sentence
|--------------------------------------------------------------------------
|
*/

function remove(string) {
   let x = 0;
   let str = string.split("").reverse();
   for (let i = 0; i < str.length; i++) {
      if (str[i] != "!") break;
      x += 1;
   }
   return string.slice(0, string.length - x);
}


function remove(s) {
   return s.replace(/!+$/, "");
}


function remove(s) {
   while (s && s.slice(-1) == "!") {
      s = s.slice(0, -1);
   }
   return s;
}


function remove(s) {
   for (let i = 0; s.endsWith("!"); i++) {
      s.endsWith("!") ? (s = s.slice(0, -1)) : s;
   }
   return s;
}


console.log(remove("Hi!"), "Hi");
console.log(remove("Hi!!!"), "Hi");
console.log(remove("!Hi"), "!Hi");
console.log(remove("!Hi!"), "!Hi");
console.log(remove("Hi! Hi!"), "Hi! Hi");
console.log(remove("Hi"), "Hi");